generic3g/CMakeLists.txt

```
@@ -58,6 +57,32 @@ esma_add_library(${this}
   DEPENDENCIES MAPL.regridder_mgr MAPL.geom_mgr MAPL.shared MAPL.profiler MAPL.base MAPL.hconfig_utils YAFYAML::yafyaml PFLOGGER::pflogger GFTL_SHARED::gftl-shared-v2 GFTL::gftl-v2
   TYPE SHARED
   )
+
+##### New function to avoid conflicts with files with the same name
+function(mapl_add_fortran_submodules)
+  set(options)
+  set(oneValueArgs TARGET)
+  set(oneValueArgs SUBDIRECTORY)
+  set(multiValueArgs SOURCES)
+  cmake_parse_arguments(
+    ARG "${options}" "${oneValueArgs}"
+    "${multiValueArgs}" ${ARGN}
+  )
+
+  foreach(file ${ARG_SOURCES})
+    set(input ${ARG_SUBDIRECTORY}/${file})
+    set(output ${ARG_SUBDIRECTORY}_${file})
+    add_custom_command(
+      OUTPUT ${output}
+      COMMAND ${CMAKE_COMMAND} -E copy ${input} ${output}
+    )
+    set_property(SOURCE ${CMAKE_CURRENT_BINARY_DIR}/${output} PROPERTY GENERATED 1)
+    target_sources(mylib PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/${output})
+  endforeach()
```

Debugged the macro more fully in my standalone sandbox
```suggestion
  foreach(file ${ARG_SOURCES})

    set(input ${CMAKE_CURRENT_SOURCE_DIR}/${ARG_SUBDIRECTORY}/${file})
    set(output ${CMAKE_CURRENT_BINARY_DIR}/${ARG_SUBDIRECTORY}_${file})
    add_custom_command(
      OUTPUT ${output}
      COMMAND ${CMAKE_COMMAND} -E copy ${input} ${output}
    )
    set_property(SOURCE ${output} PROPERTY GENERATED 1)
    target_sources(${ARG_TARGET} PRIVATE ${output})

  endforeach()
```